arcgis-ios-sdk-samples/Maps/Manage operational layers/ManageMapLayersViewController.swift

```diff
@@ -15,51 +15,88 @@
+        (navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = [
+            "ManageMapLayersViewController",
+            "MMLLayersViewController"
+        ]
                 
-        self.map = AGSMap(basemap: .topographic())
+        let map = AGSMap(basemap: .topographic())
+        map.initialViewpoint = AGSViewpoint(center: AGSPoint(x: -133e5, y: 45e5, spatialReference: .webMercator()), scale: 2e7)
         
-        let imageLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer")!)
-        self.map.operationalLayers.add(imageLayer)
+        let elevationImageLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Elevation/WorldElevations/MapServer")!)
+        let censusTiledLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")!)
         
-        let tiledLayer = AGSArcGISMapImageLayer(url: URL(string: "https://sampleserver5.arcgisonline.com/arcgis/rest/services/Census/MapServer")!)
-        self.map.operationalLayers.add(tiledLayer)
-
-        self.deletedLayers = [AGSLayer]()
+        allLayers = [elevationImageLayer, censusTiledLayer]
         
-        //add the source code button item to the right of navigation bar
-        (self.navigationItem.rightBarButtonItem as! SourceCodeBarButtonItem).filenames = ["ManageMapLayersViewController", "MMLLayersViewController"]
+        // load all the layers into the map to start
+        map.operationalLayers.addObjects(from: allLayers)
         
-        self.mapView.map = map
-        self.mapView.setViewpoint(AGSViewpoint(center: AGSPoint(x: -133e5, y: 45e5, spatialReference: AGSSpatialReference(wkid: 3857)), scale: 2e7))
+        mapView.map = map
     }
     
     //MARK: - Navigation
     
     override func prepare(for segue: UIStoryboardSegue, sender: Any?) {
-        if segue.identifier == "LayersSegue" {
-            
-            let navigationController = segue.destination as! UINavigationController
-            let controller = navigationController.viewControllers[0] as! MMLLayersViewController
-            controller.layers = self.map.operationalLayers
-            controller.deletedLayers = self.deletedLayers
-            controller.preferredContentSize = CGSize(width: 300, height: 300)
-            controller.delegate = self
+        if let navController = segue.destination as? UINavigationController,
+            let controller = navController.viewControllers.first as? MMLLayersViewController {
+            // convert and assign the map's operational layers as a Swift array
+            controller.operationalLayers = mapView.map?.operationalLayers as? [AGSLayer] ?? []
+            controller.removedLayers = removedLayers
+            controller.onChange = {[weak self] layers in
```

Comment:
Only one change can happen at a time, yes? Trying to handle all of them at the same time is an interesting idea, but it can be error prone and hard to understand. I had a hard time following what was going on. What would you think of passing the type of change, either as a parameter or by using separate callbacks per change type? I'm thinking something along the lines of `NSFetchedResultsControllerDelegate.controller(_:didChange:at:for:newIndexPath:)`. You could incorporate the index information into the change type itself:

```swift
enum ChangeType {
    case insert(index: Int)
    case delete(index: Int)
    case move(oldIndex: Int, newIndex: Int)
}
```

I'm really not trying to cause you extra work for no reason. The whole point of this sample is to show users "how to add, remove or reorder operational layers in a map". We want that to be as clear as possible. The way it is now, I don't think it is very clear.

Reply:
This is why before I just removed all the layers every time.

Reply:
That said, I don't love either of the implementations. I might revert some of them.